brainglobe_atlasapi/atlas_generation/atlas_scripts/perens_stereotaxic_mri_mouse.py

```diff
@@ -0,0 +1,325 @@
+__version__ = "0"
+import json
+import multiprocessing as mp
+import os
+import time
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+import py7zr
+from brainglobe_utils.IO.image import load_any
+from rich.progress import track
+
+from brainglobe_atlasapi import BrainGlobeAtlas, utils
+from brainglobe_atlasapi.atlas_generation.mesh_utils import (
+    Region,
+    create_region_mesh,
+)
+from brainglobe_atlasapi.atlas_generation.wrapup import wrapup_atlas_from_data
+from brainglobe_atlasapi.structure_tree_util import get_structures_tree
+
+atlas = BrainGlobeAtlas("allen_mouse_25um")
+PARALLEL = True  # disable parallel mesh extraction for easier debugging
+
+HEADERS = {
+    "User-Agent": (
+        "Mozilla/5.0 "
+        "(Windows NT 10.0; "
+        "Win64; x64; "
+        "rv:129.0) "
+        "Gecko/20100101 "
+        "Firefox/129.0"
+    ),
+    "Accept": (
+        "text/html,"
+        "application/xhtml+xml,"
+        "application/xml;q=0.9,"
+        "image/avif,"
+        "image/webp,"
+        "image/png,"
+        "image/svg+xml,"
+        "*/*;q=0.8"
+    ),
+    "Accept-Language": "en-GB,en;q=0.5",
+    "Accept-Encoding": "gzip, deflate, br, zstd",
+    "DNT": "1",
+    "Sec-GPC": "1",
+    "Host": "www.neuropedia.dk",
+    "Connection": "keep-alive",
+    "Upgrade-Insecure-Requests": "1",
+    "Sec-Fetch-Dest": "document",
+    "Sec-Fetch-Mode": "navigate",
+    "Sec-Fetch-Site": "none",
+    "Sec-Fetch-User": "?1",
+    "TE": "trailers",
+    "Priority": "u=0, i",
+}
+
+
+def download_and_extract_files(ATLAS_FILE_URL, destination_path):
+    """This is needed to get the brainglobe data from their server,
+    and bypass cloudflare which is only allowing browser access"""
+    req = urllib.request.Request(ATLAS_FILE_URL, headers=HEADERS)
+    with (
+        urllib.request.urlopen(req) as response,
+        open(destination_path, "wb") as out_file,
+    ):
+        data = response.read()  # a `bytes` object
+        out_file.write(data)
+    with py7zr.SevenZipFile(destination_path, mode="r") as z:
+        z.extractall(path=atlas_files_dir)
+
+
+### Additional functions #####################################################
+
+
+def tree_traverse_child2parent(df, child_id, ids):
+    parent = df["parent_id"][df["id"] == child_id].item()
+
+    if not np.isnan(parent):
+        id = df["id"][df["id"] == parent].item()
+        ids.append(id)
+        tree_traverse_child2parent(df, parent, ids)
+        return ids
+    else:
+        return ids
+
+
+def get_all_parents(atlas, key):
+    """
+    Get all parent IDs/acronyms in Allen's brain atlas hierarchical structure'
+
+    Call:
+        get_all_children(df, key)
+
+    Args:
+        key (int/string)      : atlas region ID/acronym
+
+    Returns:
+        parents (list) : brain region acronym corresponding to input ID
+    """
+    parents = atlas.get_structure_ancestors(temp_id)
```

Comment:
```suggestion
parents = atlas.get_structure_ancestors(key)
```
I think?

Also, not sure these two lines are worth an additional function? It's only used once, and could just be an extra line in the `for` loop to reduce levels of abstraction?